hyperspy/io_plugins/semper_unf.py

```diff
@@ -0,0 +1,520 @@
+                    label[100:100+len(self.title)] = [ord(s) for s in self.title]
+                    label[244:248] = [ord(c) for c in self.units[0]] + [0]*(4-len(self.units[0]))
+                    label[248:252] = [ord(c) for c in self.units[1]] + [0]*(4-len(self.units[1]))
+                    label[252:256] = [ord(c) for c in self.units[2]] + [0]*(4-len(self.units[2]))
+                    # Write label:
+                    f.write(struct.pack('I', 2*256))  # record length, 4 byte format!
+                    for element in label:
+                        f.write(struct.pack('h', element))  # 2 byte format!
+                    f.write(struct.pack('I', 2*256))  # record length!
+            # Write picture data:
+            for k in range(nlay):
+                for j in range(nrow):
+                    row = self.data[k, j, :]
+                    factor = 8 if self.iform == 3 else 4  # complex numbers need more space!
+                    f.write(struct.pack('I', factor*ncol))  # record length, 4 byte format!
+                    if self.iform == 0:  # bytes:
+                        raise Exception('Byte data is not supported! Use int, float or complex!')
+                    elif self.iform == 1:  # int:
+                        for element in row:
+                            f.write(struct.pack('i', element))  # 4 bytes per data entry!
+                    elif self.iform == 2:  # float:
+                        for element in row:
+                            f.write(struct.pack('f', element))  # 4 bytes per data entry!
+                    elif self.iform == 3:  # complex:
+                        for element in row:
+                            f.write(struct.pack('f', element.real))  # 4 bytes per data entry!
+                            f.write(struct.pack('f', element.imag))  # 4 bytes per data entry!
+                    f.write(struct.pack('I', factor*ncol))  # record length, 4 byte format!
+
+    @classmethod
+    def from_signal(cls, signal):
+        """Import a :class:`~.SemperFormat` object from a :class:`~hyperspy.signals.Signal` object.
+
+        Parameters
+        ----------
+        signal: :class:`~hyperspy.signals.Signal`
+            The signal which should be imported.
+
+        Returns
+        -------
+        None
+
+        """
+        data = signal.data
+        assert len(data.shape) <= 3, 'Only up to 3-dimensional datasets can be handled!'
+        scales, offsets, units = [1., 1., 1.], [0., 0., 0.], ['', '', '']
+        for i in range(len(data.shape)):
+            scales[i] = signal.axes_manager[i].scale
+            offsets[i] = signal.axes_manager[i].offset
+            units[i] = signal.axes_manager[i].units
+            if str(units[i]) == '<undefined>':
+                units[i] = ''   # Traits can not be written!
+        for i in range(3 - len(data.shape)):  # Make sure data is 3D!
+            data = np.expand_dims(data, axis=0)

```

Comment:
I would instead write [(newaxis)](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#numpy.newaxis)
```python
data = data[tuple(None for _ in xrange(3-len(data.shape)))]
```
alternatively `np.atleast_3d ` may also be used, but I guess you want to extend the dimensions only in particular direction?